pkg/rancher-desktop/pages/Containers.vue

```diff
@@ -331,25 +331,27 @@ export default {
       return { content: sha };
     },
     getUniquePorts(ports) {
-      const keys = Object.keys(ports);
+      const displayMap = [];
 
-      const uniquePortMap = keys.map((key) => {
-        const values = ports[key];
-        const hostPorts = values.map(value => value.HostPort);
-        const uniqueHostPorts = [...new Set(hostPorts)];
+      Object.entries(ports).forEach((entry) => {
+        const [key, values] = entry;
 
-        return { [key]: uniqueHostPorts };
-      });
-
-      const displayMap = uniquePortMap.map((element) => {
-        const key = Object.keys(element)[0];
-        const values = element[key];
-        const port = key.split('/')[0];
-
-        return values.map(value => `${ value }:${ port }`);
+        if (!values) {
+          return;
+        }
+        try {
+          const hostPorts = (values || []).map(value => value.HostPort);
```

Comment:
`values || []` is redundant because `if (!values) { return }` above. We can probably drop the early return?

Reply:
Ah right. The reason for the failure was that some of the port settings in the mocked container data didn't have a target, and the code was looping over null. So I added the `|| []` alternative first and then checked out what an early return would look like. But it isn't needed.